Tests/MongoSwiftTests/TestUtils.swift

```diff
@@ -37,10 +38,21 @@ class MongoSwiftTestCase: XCTestCase {
     /// If the environment variable does not exist, this will use a default of "mongodb://127.0.0.1/".
     static var connStr: String {
         if let connStr = ProcessInfo.processInfo.environment["MONGODB_URI"] {
+            if topologyType == .sharded {
+                let uri = mongoc_uri_new(connStr)
+                let hosts = mongoc_uri_get_hosts(uri)!
+                let hostAndPort = withUnsafeBytes(of: hosts.pointee.host_and_port) { rawPtr -> String in
+                    let ptr = rawPtr.baseAddress!.assumingMemoryBound(to: CChar.self)
```

Comment:
baseAddress can be nil if the buffer is empty, but maybe we are confident enough that libmongoc filled this out that we don't care?

Reply:
added a few guards, I think it's unlikely we can get to that state anymore